middlewares/stripPrefixRegex_test.go

```diff
@@ -73,18 +73,16 @@ func TestStripPrefixRegex(t *testing.T) {
 				actualPath = r.URL.Path
 				actualHeader = r.Header.Get(ForwardedPrefixHeader)
 			})
-
 			handler := NewStripPrefixRegex(handlerPath, testPrefixRegex)
-			server := httptest.NewServer(handler)
-			defer server.Close()
 
-			resp, err := http.Get(server.URL + test.path)
-			require.NoError(t, err, "%s: unexpected error.", test.path)
+			req := testhelpers.MustNewRequest(http.MethodGet, "http://localhost"+test.path, nil)
+			resp := &httptest.ResponseRecorder{Code: http.StatusOK}
+
+			handler.ServeHTTP(resp, req)
 
-			assert.Equal(t, test.expectedStatusCode, resp.StatusCode, "%s: unexpected status code.", test.path)
+			assert.Equal(t, test.expectedStatusCode, resp.Code, "%s: unexpected status code.", test.path)
```

Comment:
I need to do that because I don't see the context case (sub-test) when a test failed.

```
/home/ldez/.gvm/gos/go1.8.3/bin/go test -c -i -o /tmp/TestStripPrefixRegex_in_stripPrefixRegex_test_gogo github.com/containous/traefik/middlewares
/tmp/TestStripPrefixRegex_in_stripPrefixRegex_test_gogo -test.v -test.run ^TestStripPrefixRegex$
	Error Trace:	stripPrefixRegex_test.go:83
	Error:      	Not equal: 
	            	expected: 201
	            	received: 200
	Messages:   	/b/api2/test2: unexpected status code.
	Error Trace:	stripPrefixRegex_test.go:83
	Error:      	Not equal: 
	            	expected: 405
	            	received: 404
	Messages:   	/a/test: unexpected status code.
	Error Trace:	stripPrefixRegex_test.go:83
	Error:      	Not equal: 
	            	expected: 201
	            	received: 200
	Messages:   	/b/api/: unexpected status code.
	Error Trace:	stripPrefixRegex_test.go:83
	Error:      	Not equal: 
	            	expected: 201
	            	received: 200
	Messages:   	/c/api/123/test3: unexpected status code.
	Error Trace:	stripPrefixRegex_test.go:83
	Error:      	Not equal: 
	            	expected: 201
	            	received: 200
	Messages:   	/c/api/123/: unexpected status code.
	Error Trace:	stripPrefixRegex_test.go:83
	Error:      	Not equal: 
	            	expected: 201
	            	received: 200
	Messages:   	/a/api/test: unexpected status code.
	Error Trace:	stripPrefixRegex_test.go:83
	Error:      	Not equal: 
	            	expected: 405
	            	received: 404
	Messages:   	/c/api/abc/test4: unexpected status code.
	Error Trace:	stripPrefixRegex_test.go:83
	Error:      	Not equal: 
	            	expected: 201
	            	received: 200
	Messages:   	/b/api/test1: unexpected status code.

```